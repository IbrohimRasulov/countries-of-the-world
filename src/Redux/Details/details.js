const GET_DETAILS_API = 'CountriesOfTheWorld/details/GET_DETAILS';


const initialState = [];

export const getDetails = (payload) => ({
  type: GET_DETAILS_API,
  payload,
});

export const getDetailsFromAPI = (country) => async (dispatch) => {
  await fetch(`https://restcountries.com/v2/name/${country}`)
    .then((response) => response.json())
    .then((data) => {
      const arrData = data.map((country) => ({
        name: country.name,
        flag: country.flags.png,
        region: country.region,
        callingCodes: country.callingCodes,
        population: country.population,
        area: country.area,
        timezones: country.timezones[0],
        currencies: country.currencies.name,
        languages: country.languages.name
      }));
      if (arrData) {
        dispatch(getDetails(arrData));
      }
    });
};

const detailsReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_DETAILS_API:
      return [...action.payload];
    default:
      return state;
  }
};

export default detailsReducer;
