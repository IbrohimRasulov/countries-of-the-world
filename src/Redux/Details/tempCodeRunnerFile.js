https://restcountries.com/v2/name/${country}