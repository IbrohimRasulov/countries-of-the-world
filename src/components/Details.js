import Header from "./Header";
import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { getDetailsFromAPI } from '../Redux/Details/details';
import { useParams } from 'react-router-dom';

const Details = () => {
  const countryDetails = useSelector((state) => state.detailsReducer)
  const dispatch = useDispatch();
  const params = useParams();


  const { countryName } = params;

  useEffect(() => {
    dispatch(getDetailsFromAPI(countryName));
  }, [dispatch, countryName]);

  // console.log(countryDetails);

  return (
    <div className="details">
      <Header className="App-header" search="false" />
    </div>
  );
}

export default Details;
