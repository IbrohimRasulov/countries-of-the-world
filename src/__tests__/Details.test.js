import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../Redux/configureStore';
import Details from '../components/Details';

it('testing rendering Details component', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <Details />
      </Router>
    </Provider>,
  ).toJSON;
  expect(tree).toMatchSnapshot();
});
