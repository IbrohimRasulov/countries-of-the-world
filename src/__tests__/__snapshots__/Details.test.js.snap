// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing rendering Details component 1`] = `[Function]`;

exports[`testing rendering Profile component 1`] = `[Function]`;
